libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/DefaultSessionPreferencesStore.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.preferences.impl.store
+
+import android.content.Context
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStoreFile
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.features.preferences.api.store.SessionPreferencesStore
+import io.element.android.libraries.di.ApplicationContext
+import io.element.android.libraries.di.SessionScope
+import io.element.android.libraries.di.SingleIn
+import io.element.android.libraries.matrix.api.user.CurrentSessionIdHolder
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+@ContributesBinding(SessionScope::class)
+@SingleIn(SessionScope::class)
+class DefaultSessionPreferencesStore @Inject constructor(
+    @ApplicationContext context: Context,
+    currentSessionIdHolder: CurrentSessionIdHolder,
+) : SessionPreferencesStore {
+    private val privateReadReceiptsKey = booleanPreferencesKey("privateReadReceipts")
+
+    private val store = PreferenceDataStoreFactory.create {
+        context.preferencesDataStoreFile("session_${currentSessionIdHolder.current}_preferences")
```

Comment:
You have to use a hash of the user id here, because it will crash if the user name is too long. See #2237

Reply:
https://github.com/element-hq/element-x-android/pull/2290/commits/da6c11ab0058741937ebe2324c712da03a0f1824